ExternalModule.php

```diff
@@ -164,7 +164,10 @@ function setDefaultValues() {
 
                     // Getting previous event ID.
                     foreach ($events as $event) {
-                        if ($event == $_GET['event_id']) {
+                        // Event instances share the same event id for every instance. 
+                        // To support event instances we only break on the first instance, and allow execution for all future instances i.e., $_GET['instance'] greater than 1. 
+                        // Event instances are enabled under: Project Setup > Enable optional modules and customizations > Repeatable instruments and events > 'Repeat instruments (repeat independently of each other)`
+                        if ($event == $_GET['event_id'] && $_GET['instance'] == 1) {
```

Comment:
This is the cause of the event-id mismatch. The bug I'm seeing results from this `foreach` running to the end of the `$events`, resulting the final event being set as `$prev_event`